File: android/src/main/kotlin/com/ecommpay/flutterplugin/models/PluginPaymentInfo.kt
package com.ecommpay.flutterplugin.models

import com.ecommpay.msdk.ui.EcmpPaymentInfo
import kotlinx.serialization.Serializable

@Serializable
internal data class PluginPaymentInfo(
    val projectId: Int,
    val paymentId: String,
    val paymentAmount: Long,
    val paymentCurrency: String,
    val paymentDescription: String?,
    val customerId: String?,
    val regionCode: String?,
    val token: String?,
    val receiptData: String?,
    val languageCode: String?,
    val hideSavedWallets: Boolean?,
    val forcePaymentMethod: String?,
    val signature: String?,
) {
    fun map() = EcmpPaymentInfo(
        projectId = this.projectId,
        paymentId = this.paymentId,
        paymentAmount = this.paymentAmount,
        paymentCurrency = this.paymentCurrency,
        //optional fields
        paymentDescription = this.paymentDescription,
        customerId = this.customerId,
        regionCode = this.regionCode,
        token = this.token,
        languageCode = this.languageCode,
        receiptData = this.receiptData,
        hideSavedWallets = this.hideSavedWallets ?: false,
        forcePaymentMethod = this.forcePaymentMethod,
        signature = this.signature,
    )
}

File: ios/Classes/Models/PluginPaymentInfo.swift
//
//  PluginPaymentInfo.swift
//  ecmpplugin
//
//  Created by Alex on 18.04.2024.
//

import Foundation
import ecommpaySDK

internal struct PluginPaymentInfo: Decodable {
    let projectId: Int32
    let paymentId: String
    let paymentAmount: Int64
    let paymentCurrency: String
    let paymentDescription: String?
    let customerId: String?
    let regionCode: String?
    let token: String?
    let languageCode: String?
    let receiptData: String?
    let forcePaymentMethod: String?
    let hideSavedWallets: Bool?
    let signature: String?
    
    func map() -> PaymentOptions {
        let paymentOptions = PaymentOptions(projectID: projectId,
                                            paymentID: paymentId,
                                            paymentAmount: paymentAmount,
                                            paymentCurrency: paymentCurrency,
                                            paymentDescription: paymentDescription,
                                            customerID: customerId, // unique ID assigned to your customer
                                            regionCode: regionCode,
                                            token: token)
        paymentOptions.receiptData = receiptData
        paymentOptions.languageCode = languageCode
        paymentOptions.forcePaymentMethod = forcePaymentMethod
        paymentOptions.hideSavedWallets = hideSavedWallets ?? false
        paymentOptions.signature = signature
        
        return paymentOptions
    }
}
